pkg/kapp/diffgraph/placeholder_parser.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diffgraph
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	ctlres "github.com/k14s/kapp/pkg/kapp/resources"
+	ctlcrd "github.com/k14s/kapp/pkg/kapp/resourcesmisc"
+)
+
+type PlaceholderParser struct {
```

Comment:
in a context of diffgraph PlaceholderParser is a bit too general of a name. how about:

```
PlaceholderParser                -> ChangeGroupName
NewPlaceholderParserWithResource -> NewChangeGroupNameForResource(name, res) => ChangeGroupName
PlaceholderParser.Parse(...)     -> ChangeGroupName.AsString() => (name, error)
```

(placeholder is more of an implementation detail)

i would also recommend shifting valueMapFromResource call into AsString (currently Parse), away from NewChangeGroupNameForResource as there are benefits of not doing any work in constructors (aka delayed execution). in our current usage the only benefit we get is that we can drop parseChangeGroupPlaceholders, and just use NewChangeGroupNameForResource(name, res).AsString().

Reply:
Refactored `placeholder_parser` to `change_group_name.go` keeping in mind the recommended changes